app/components/blacklight/gallery/slideshow_preview_component.rb

```diff
@@ -7,7 +7,14 @@ class SlideshowPreviewComponent < Blacklight::DocumentComponent
 
       def initialize(document:, document_counter: nil, **args)
         super(document: document, document_counter: document_counter, **args)
-        @document_counter = document_counter || @counter
+        # ViewComponent 3, document_counter = 0
+        # ViewComponent 3, @counter = 1
+        # ViewComponent 3, COLLECTION_INDEX_OFFSET = 1
+
+        # ViewComponent 2, document_counter = 1
```

Comment:
Can you add a view_component 2 and 3 configuration to the CI config so that both ways are adequately tested?  Do we still need to support VC 2?

Reply:
We have apps still using VC2. We want to continue to support.